dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunction.cs

```diff
@@ -308,31 +298,32 @@ internal KernelFunction(string name, string? pluginName, string description, IRe
                 throw;
             }
 
-            // Ensure we clean up after the enumerator.
-            await using (enumerator.ConfigureAwait(false))
+            if (enumerator is not null)
```

Comment:
It is not expected to have enumerator as null, but in this implementation enumerator is assigned in filter callback, so when I define an enumerator I need to assign some initial value to it, otherwise there will be a warning that I'm accessing unassigned variable.

Reply:
> otherwise there will be a warning that I'm accessing unassigned variable.

If there's no way it could be null, you can just use `!` to suppress the warning. (Ideally instead you'd use Debug.Assert to suppress it, but that won't do so in a netstandard2.0 build.)